[dynamic registrar] Out parameters must be released.

rolfbjarne opened the pull request:
Before:

    There were 258042 MonoObjects created, 235166 MonoObjects freed, so 22876 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204219 MonoObjects freed, so 1585 were not freed. (static registrar)

After:

    There were 258058 MonoObjects created, 235308 MonoObjects freed, so 22750 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204219 MonoObjects freed, so 1585 were not freed. (static registrar)

rolfbjarne commented:
/azp run

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812978&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/de24597c9500cf2246619ac643bc3005)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/409149f892d9039e4c17e2d61b8f0522)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/9a365afa528ddd48e76d6423d912a5bb)
* macCatiOS  [gist](https://gist.github.com/0527844c6bdca02dbfdb27a0fa342b05)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/d43001062296f5fddfadf5a8cf196ec1)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/4f39450442a3349d675d4196a2d833ac)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.18/4812978/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11727/4812978/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11727/4812978/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11727/4812978/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 106 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Failed (Test run failed.
Tests run: 2444 Passed: 2353 Inconclusive: 7 Failed: 4 Ignored: 87)
 * monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Failed (Test run failed.
Tests run: 2441 Passed: 2353 Inconclusive: 7 Failed: 2 Ignored: 86)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812978&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1101.BigSur'
Merge dd3ae7fb1febd7ee763a6af4ef18c4a6358d8027 into 51c936457d6ce0f7f596179f0fbe6b294c56f6c7

rolfbjarne commented:
Test failures are unrelated

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Fixed in #11732, #11733 and #11734.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Fixed in #11732 and #11733.
